### ISSUE TITLE
Address Mapping

### ISSUE THREAD

**Issue by gtxzsxxk** [Low-Speed-Linux-Experimental-Platform/temu#5]
Our memory are very rare, so

```
0x0000_0000: ROM START

0x0200_0000: RAM START
290KiB OpenSBI

0x0204_8800: FDT
10KiB

0x0204_B000: U-boot (Kernel decompress start)
512KiB U-boot

0x020c_B000: Kernel decompress space
8MiB

0x028c_B000: Compressed Kernel at
3MiB

0x02bc_B000: Uboot reloc

0x0300_0000: RAM END 16MiB
```

**Comment by gtxzsxxk** [comment]
```
unzip 0x028cb000 0x0204b000
go 0x0204b000
```

**Comment by gtxzsxxk** [comment]
```
unzip 0x028cb000 0x0204afc0
bootm 0x0204afc0 - 0x02f9b830
```

**Comment by gtxzsxxk** [comment]
```
mkimage -A riscv -O linux -T kernel -C none -a 0x0204b000 -e 0x0204b000 -d arch/riscv/boot/Image uImage
```

**Comment by gtxzsxxk** [comment]
```
0x0000_0000: ROM START

0x0200_0000: RAM START
290KiB OpenSBI

0x0204_8800: FDT
10KiB

0x0204_B000: U-boot 
512KiB U-boot

0x0210_0000: Kernel Decompress Start

0x020c_B000: Kernel decompress space
8MiB

0x028c_B000: Compressed Kernel at
3MiB

0x02bc_B000: Uboot reloc

0x0300_0000: RAM END 16MiB
```
unzip 0x028cb000 0x020fffc0
bootm 0x020fffc0 - 0x02f9b830
```
```

**Comment by gtxzsxxk** [comment]
latest

```
0x0000_0000: ROM START

0x0200_0000: RAM1 START
290KiB OpenSBI

0x0204_8800: FDT
10KiB

0x0204_B000: U-boot 
512KiB U-boot

0x028c_B000: Compressed Kernel at
3MiB

0x02bc_B000: Uboot reloc

0x0300_0000: RAM1 END 16MiB all used 5MiB

0x8000_0000: RAM2 start (Kernel Decompress Start)
10MiB
0x80a0_0000: RAM2 end
```

```
unzip 0x028cb000 0x8000000
bootm 0x8000000 - 0x02f9b830
```

```
mkimage -A riscv -O linux -T kernel -C none -a 0x02100000 -e 0x02100000 -d arch/riscv/boot/Image uImage
```

**Comment by gtxzsxxk** [comment]
```
riscv32-buildroot-linux-gnu-objdump vmlinux > vmlinux.dump -D -S -M no-aliases,numeric
```

**Comment by gtxzsxxk** [comment]
We must start at 0x8000_0000. Here

```
0x0000_0000: ROM START

0x8000_0000: U-boot, (Kernel Decompress Start)
512KiB U-boot

0x80ab_5000: Compressed Kernel at
3MiB

0x80db_5000:  OpenSBI
290KiB

0x80df_d800: FDT
10KiB

0x80e0_0000: Uboot reloc
2MiB

0x8100_0000: RAM end 16MiB
```

```
unzip 0x028cb000 0x80000000
setenv bootargs "earlycon=sbi"
bootm 0x80000000 - 0x02048800
```

```
mkimage -A riscv -O linux -T kernel -C none -a 0x80000000 -e 0x80000000 -d arch/riscv/boot/Image uImage
```

Available memory for kernel: **13MiB**

**Comment by gtxzsxxk** [comment]
We must start at 0x8000_0000. Here

**Enlarge everything to 32MiB**

```
0x0000_0000: ROM START

0x8000_0000: U-boot, (Kernel Decompress Start)
512KiB U-boot

0x81ab_5000: Compressed Kernel at
3MiB

0x81dc_0000:  OpenSBI
290KiB

0x81df_d800: FDT
10KiB

0x81e0_0000: Uboot reloc
2MiB

0x8200_0000: RAM end 32MiB
```

```
unzip 0x81ab5000 0x80000000
setenv bootargs "earlycon=sbi"
bootm 0x80000000 - 0x81dfd800
```

```
mkimage -A riscv -O linux -T kernel -C none -a 0x80000000 -e 0x80000000 -d arch/riscv/boot/Image uImage
```

Available memory for kernel: **29MiB**

**Comment by gtxzsxxk** [comment]
```
unzip 0x80aa0000 0x80000000
setenv bootargs "earlycon=sbi console=ttyS0,115200n8"
bootm 0x80000000 - 0x80ffd800
```

**Comment by gtxzsxxk** [comment]
```
setenv bootargs "earlycon=uart8250,mmio32,0x12500000,115200n8 console=ttyS0,115200n8"
```

**Comment by gtxzsxxk** [comment]
setenv bootargs "console=tty0 console=ttyS0,115200"

**Comment by gtxzsxxk** [comment]
```
setenv bootargs "console=ttyS0,115200"
```

**Comment by gtxzsxxk** [comment]
## Latest

```
0x0000_0000: ROM START

0x8000_0000: U-boot, (Kernel Decompress Start)
512KiB U-boot

0x808a_0000: Compressed Kernel at
5MiB

0x80da_0000: Uboot reloc
2MiB

0x80fa_0000:  OpenSBI
384KiB

0x80ff_d800: FDT
10KiB

0x8100_0000: RAM end 16MiB
```

```
unzip 0x80aa0000 0x80000000
setenv bootargs "earlycon=sbi"
bootm 0x80000000 - 0x80ffd800
```

```
mkimage -A riscv -O linux -T kernel -C none -a 0x80000000 -e 0x80000000 -d arch/riscv/boot/Image uImage
```

Available memory for kernel: **13.625MiB**

```
--addr=0x80fa0000
--exec=fw_jump.bin
--with=0x80000000#u-boot.bin
--with=0x80ffd800#u-boot.dtb
--with=0x80aa0000#uImage.gz
```

**Comment by gtxzsxxk** [comment]
```
setenv bootargs earlycon=sbi,null console=ttyS0,115200 root=/dev/ram0
```
